analyze.py

```diff
@@ -374,13 +375,23 @@ def analyze(audiofile, outfile, compute_timbral_models=False, compute_descriptor
     parser.add_argument('-t', '--timbral-models', help='include descriptors computed from timbral models', action='store_const', const=True, default=False)
     parser.add_argument('-m', '--music-pieces', help='include descriptors designed for music pieces', action='store_const', const=True, default=False)
     parser.add_argument('-s', '--music-samples', help='include descriptors designed for music samples', action='store_const', const=True, default=False)
-    parser.add_argument('-i', '--input', help='input audio file', required=True)
-    parser.add_argument('-o', '--output', help='output analysis file', required=True)
+    parser.add_argument('-i', '--input', help='input audio file or input directory containing the audio files to analyze', required=True)
+    parser.add_argument('-o', '--output', help='output analysis file or output directory where the analysis files will be saved', required=True)
     parser.add_argument('-f', '--format', help='format of the output analysis file ("json" or "jsonld", defaults to "jsonld")', default="jsonld")
     parser.add_argument('-u', '--uri', help='URI for the analyzed sound (only used if "jsonld" format is chosen)', default=None)
     
     args = parser.parse_args()
     logging.basicConfig(format='%(asctime)s %(levelname)s:%(message)s', level=logging.INFO if not args.verbose else logging.DEBUG)
 
-    analyze(args.input, args.output, args.timbral_models, args.music_pieces, args.music_samples, args.format, args.uri)
-    
+    # check if the input is a file or a folder
+    if os.path.isfile(args.input):
+        analyze(args.input, args.output, args.timbral_models, args.music_pieces, args.music_samples, args.format, args.uri)
+    # ensure that the output argument is a folder
+    elif not os.path.isfile(args.output):
```

Comment:
this check also returns False if the output folder doesn't exist, which will cause problems. Do you want this folder to be created by the user before they call this, or should the script create it?
Perhaps it'd be better to change the order in which you do these checks, so that you first check if both input and output are a directory, and do that first:

```py
if os.path.isdir(input) and os.path.isdir(output):
    ... process as a dir
elif os.path.isfile(input): # optional not os.path.exists(output)
    ... process as a file
```

Reply:
Good points. 
The output folder might be created automatically if the user mounts the folders as proposed in the Readme:
```
docker run -it --rm -v /local/path/to/your/input_directory/:/audio -v /local/path/to/output_directory/:/outdir mtgupf/ac-audio-extractor:v3 -i /audio/ -o /outdir/  -st
```

If the users mounts things diferently (e.g. mounts only one folder containing the output folder in which she wants to store the analysis), an exception will be raised telling the user to `Make sure input and output arguments are both files or folders`. I think it is enough so that the user can check that the output folder exists in the case of analyzing all the files within a folder.